src/core/manager.c

```diff
@@ -2519,21 +2520,19 @@ int manager_deserialize(Manager *m, FILE *f, FDSet *fds) {
         m->n_reloading ++;
 
         for (;;) {
-                char line[LINE_MAX], *l;
+                _cleanup_free_ char *line = NULL;
+                char *l;
 
-                if (!fgets(line, sizeof(line), f)) {
-                        if (feof(f))
-                                r = 0;
-                        else
-                                r = -errno;
 
-                        goto finish;
-                }
+                r = read_line(f, LONG_LINE_MAX, &line);
+                if (r < 0)
+                        return log_error_errno(r, "Failed to read serialization line: %m");
```

Comment:
Eh. We screwed this up in the original backport. If this `return` ever happens, `m->n_reloading` will remain >0, with unclear consequences (but I doubt they will be positive). The `return` needs to be changed to `goto finish`.